variables.tf

```diff
@@ -16,8 +22,48 @@ variable "tags" {
 
 variable "compute_environments" {
   description = "Map of compute environment definitions to create"
-  type        = any
-  default     = {}
+  type = map(object({
+    name        = optional(string)
+    name_prefix = optional(string)
+    compute_resources = optional(object({
+      allocation_strategy = optional(string)
+      bid_percentage      = optional(number)
+      desired_vcpus       = optional(number)
+      ec2_configuration = optional(list(object({
+        image_id_override = optional(string)
+        image_type        = optional(string)
+      })))
+      ec2_key_pair   = optional(string)
+      instance_role  = optional(string)
+      instance_types = optional(list(string))
+      launch_template = optional(object({
+        launch_template_id   = optional(string)
+        launch_template_name = optional(string)
+        version              = optional(string)
+      }))
+      max_vcpus           = number
+      min_vcpus           = optional(number)
+      placement_group     = optional(string)
+      security_group_ids  = optional(list(string))
+      spot_iam_fleet_role = optional(string)
+      subnets             = list(string)
+      tags                = optional(map(string), {})
+      type                = string
+    }))
+    eks_configuration = optional(object({
+      eks_cluster_arn      = string
+      kubernetes_namespace = string
+    }))
+    service_role = optional(string)
+    state        = optional(string)
+    tags         = optional(map(string), {})
+    type         = optional(string, "MANAGED")
```

Comment:
Such a detailed variable type may be a good idea, but do we need to provide a default value (`"MANAGED"`) in the type definition or leave it empty, so that it will be `null` and let the AWS provider decide on the correct value? We previously left it empty for the provider to decide. I think it is still right to do.